Yank/experiment.py

```diff
@@ -280,7 +280,9 @@ def create_alchemical_phase(self):
             checkpoint_interval = self.options['checkpoint_interval']
             # Get the solute atoms
             if self.options['store_solute_trajectory']:
-                solute_atoms = self.topography.solute_atoms
+                # "Solute" is basically just not water. Includes all non-water atoms and ions
+                # Use set union (|) cast back to list to get output
+                solute_atoms = list(frozenset(self.topography.solute_atoms) | frozenset(self.topography.ions_atoms))
```

Comment:
`Topography` already guarantees that the intersection of `solute_atoms` and `ions_atoms` is null. I think we should do this with something like
```python
solute_atoms  = self.topography.solute_atoms + self.topography.ions_atoms
```
because going through a `set` can scramble the order of the atom indices, which feels more error-prone. The order should already be correct as ions atoms are always added after the solute.

Reply:
> intersection of solute_atoms and ions_atoms is null.

Ah, I missed that, 

>  going through a set can scramble the order

Which is why I used frozen set. But since they already don't intersect, I can fix that, but this seemed more future proof to me.

Reply:
I'm not sure that `frozenset` differs from the normal `set` in this. I believe there's no guarantee about the order of the elements in both cases, especially after the intersection operation.